src/app/app.component.ts

```diff
@@ -28,8 +33,22 @@ import { AuthService } from './services/auth.service';
 })
 export class AppComponent {
   title = 'bon-voyage';
-  router = inject(Router);
-  authService = inject(AuthService);
+
+  constructor(
+    private store: Store<TripState>,
+    private router: Router,
+    protected authService: AuthService
+  ) {
+    this.store
+      .select(selectBaseCurrency)
+      .pipe(takeUntilDestroyed())
+      .subscribe(baseCurrency => {
+        this.store.dispatch(
+          getExchangeRates({ baseCurrency: baseCurrency ?? 'ZAR' })
+        );
+      });
+    this.store.dispatch(getAllTrips());
+  }
 
   signOut() {
     this.authService.logUserOut();
```

Comment:
Consider moving out anything that has to deal with side effects into ngrx effects

Reply:
Will be making a user state in the future, so this is subject to change. Thanks 👍🏻